File: routes/auth.routes.go
package routes

import (
	"github.com/Tsuzat/zipit-go-fiber/config"
	"github.com/Tsuzat/zipit-go-fiber/controller"
	"github.com/Tsuzat/zipit-go-fiber/middleware"
)

func InitAuthRouter() {
	group := config.APP.Group("/api/v1/auth")

	group.Post("/signup", controller.SignUpUser)
	group.Post("/login", controller.LoginUser)
	group.Get("/me", middleware.Authenticate, controller.Me)
	group.Get("/logout", controller.LogOut)
	group.Post("/refresh-access-token", controller.RefreshToken)
	group.Get("/verify", controller.VerifyUserEmail)
}

File: src/routes/auth.routes.ts
import { Router } from "express";
import {
  loginUser,
  registerUser,
  verifyEmail,
} from "../controllers/auth.controllers";

const authRouter = Router();

authRouter.route("/signup").post(registerUser);
authRouter.route("/verify-email").get(verifyEmail);
authRouter.route("/login").post(loginUser);

export { authRouter };
